### ISSUE TITLE
Clean up usePress code

### ISSUE THREAD

**Issue by devongovett** [adobe/react-spectrum#1720]
1.  ~~We can probably get rid of the fallback events (mouse/touch) now since all supported browsers support pointer events.~~ https://github.com/adobe/react-spectrum/discussions/5636#discussioncomment-8053511 is why we can't get rid of the fall backs
2. Clean up the various states to simplify the code.
3. Possibly get rid of the custom hit testing. The code comment indicates that this was for iOS < 13.2 which is long gone, so possibly this isn't needed anymore. We will need to do some testing. Related: #1061 #1613 DONE: https://github.com/adobe/react-spectrum/pull/7427

**Comment by cpatti97100** [comment]
Hi all, is this being taken into consideration?

**Comment by snowystinger** [comment]
Yes, however, it has some nuances that we'll need to account for, see https://github.com/adobe/react-spectrum/discussions/5636#discussioncomment-8053511

feel free to make a PR for it

**Comment by nicosammito** [comment]
Would be great if a fix is introduced

**Comment by freisenhauer** [comment]
Any updates on this ? This or rather the issue that buttons cannot be used by vimium is a real issue for me. This blocks my entire workflow.

**Comment by devongovett** [comment]
Going to consider this complete. In the last few releases, we've significantly refactored `usePress` to [remove custom hit testing](https://react-spectrum.adobe.com/releases/2025-01-15.html#enhancements), [avoid preventDefault](https://react-spectrum.adobe.com/releases/2025-03-05.html#usepress-update), [reduce bundle size and support onClick as an alias](https://react-spectrum.adobe.com/releases/2025-04-11.html#enhancements).
